drpcmetadata/metadata.go

```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/zeebo/errs"
 )
 
-// AddPairs attaches metadata onto a context and return the context.
+// AddPairs attaches metadata onto an incoming context and returns the context.
 func AddPairs(ctx context.Context, metadata map[string]string) context.Context {
```

Comment:
All existing methods refer to incoming context and I have not changed the names of the existing methods as they are already being used both in the drpc and cockroach repo. We could go with the semantics that 'incoming' is the default context (although I've added NewIncomingContext for now and can rename it to NewContext if we agree with this). Let me know if you feel strongly otherwise. Thanks.